gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -427,14 +438,18 @@ private void createJobPropsForTopic(String topic, double priority) throws IOExce
    * Get an instance of {@link MRCompactorJobPropCreator}, e.g., {@link MRCompactorTimeBasedJobPropCreator}
    * for time-based compaction.
    */
-  MRCompactorJobPropCreator getJobPropCreator(String topic, Path topicInputDir, Path topicOutputDir, Path topicTmpDir) {
+
+  MRCompactorJobPropCreator getJobPropCreator(String topic) {
     String builderClassName =
         this.state.getProp(COMPACTION_JOBPROPS_CREATOR_CLASS, DEFAULT_COMPACTION_JOBPROPS_CREATOR_CLASS) + "$Builder";
     boolean deduplicate = this.state.getPropAsBoolean(COMPACTION_DEDUPLICATE, DEFAULT_COMPACTION_DEDUPLICATE);
 
     try {
       return ((MRCompactorJobPropCreator.Builder<?>) Class.forName(builderClassName).newInstance()).withTopic(topic)
-          .withTopicInputDir(topicInputDir).withTopicOutputDir(topicOutputDir).withTopicTmpDir(topicTmpDir)
+          .withTopicInputDir(new Path(this.inputDir, new Path(topic, this.inputSubDir)))
+          .withTopicOutputDir(new Path(this.destDir, new Path(topic, this.destSubDir)))
+          .withTopicOutputBaseDir(new Path(this.destDir, topic))
```

Comment:
The style is not consistent here. Either change this line to `withTopicOutputLateDir(...)`, or if you want to use `withTopicOutputBaseDir()`, then you don't need `withTopicInputDir()`, `withTopicOutputDir()` and `withTopicTmpDir()` because they can all be constructed from base dir.